packages/x-date-pickers-pro/src/MobileDateRangePicker/MobileDateRangePicker.types.ts

```diff
@@ -19,13 +19,12 @@ export interface MobileDateRangePickerSlotProps<TDate>
     UseMobileRangePickerSlotProps<TDate, 'day'> {}
 
 export interface MobileDateRangePickerProps<TDate>
-  extends BaseDateRangePickerProps<TDate>,
+  extends Omit<
+      BaseDateRangePickerProps<TDate>,
```

Comment:
Would it make sense to do the omit at the `BaseDateRangePickerProps` level ?
Given the name of this interface, I would expect it to only contain things that are shared by all variants.

Reply:
Sure, makes more sense! I missed the fact that we already re-specified the `calendars` prop in the Desktop pickers props type anyway. 🙈 👍

Reply:
Here are the necessary changes: https://github.com/mui/mui-x/pull/9528/commits/4875a16e520fab0e3fcdd586a659b5b93500a1bb

Reply:
Sorry for being super picky on this one :grimacing: 

Usually, the `ExportedXXX` interfaces are used on the pickers to decide if a prop is exported by the parent of a component or not.
So we could just move `calendars` from `ExportedDateRangeCalendarProps` to `DateRangeCalendarProps` and remove the `Omit`.

Reply:
Fair point. 👍 
I didn't do that, because I thought that the theme augmentation also uses the `Exported` type and we definitely want to keep the option of overriding the `calendars` there.
But after checking the props, it seems that we are all over the place with props that we allow to be overridden via theme `defaultProps`. 🤔

Reply:
Our theme management is probably far from ideal.
I don't think the theme overrides should be limited to `ExportedXXX` props, or at least that's not how those interfaces were designed when I migrated from the lab.
The `ExportedDateCalendarProps` interface contains everything that the pickers using it will also have as public API.
But when setting default props to the theme for `DateCalendar`, you might want to use other props that the pickers do not expose.

I'm not entirely sure this `ExportXXXX` abstraction is good, it was clearly one of my main pain point when starting to work on the pickers.

Reply:
> If we really wanted to have clean theme overrides, we would probably need another notion of `PublicXXXX` interfaces when we want to keep some props private and non-overridable (or `ThemeXXX`)

Yes, that's a good point. We might have to eventually do a thorough check if we are not overly open to extension via theme overrides and possibly create a bit more limited types if necessary. 👍